protos/events/events.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+option go_package = "events";
+
+package github.com.tinkerbell.tink.protos.events;
+
+import "timestamp.proto";
+
+service EventsService {
+  rpc Watch(WatchRequest) returns (stream Event){}
+}
+
+message Event {
+  string id = 1;
+  string resource_id = 2;
+  ResourceType resource_type = 3;
+  EventType event_type = 4;
+  bytes data = 5;
+  google.protobuf.Timestamp created_at = 6;
+}
+
+message WatchRequest {
+  EventType event_type = 1;
+  ResourceType resource_type = 2;
+
+  // Every consumer when it connects to a stream of events will specify
+  // how old the events returned should be (by default 5m).
+  int32 old_events_time_span = 3; // needs a better name
+}
+
+enum ResourceType {
+  TEMPLATE = 0;
+  HARDWARE = 1;
+  WORKFLOW = 2;
```

Comment:
What about ACTION events as they move from pending to in progress to completed / failed?

Reply:
When I think about a resource I usually think about entities stores in a database. Actions are part of a workflow. They will have their own set of events but they are not a resource in the way I intend them

Reply:
I can see having an ACTION ResourceType, it would cut down on pushing the full workflow down the event pipe when an action has been updated, but I think we can try it this way for now.